xbmc/input/linux/LinuxInputDevices.cpp

```diff
@@ -460,6 +461,13 @@ bool CLinuxInputDevice::KeyEvent(const struct input_event& levt, XBMC_Event& dev
 {
   int code = levt.code;
 
+  if((levt.code == BTN_TOUCH || levt.code == BTN_TOOL_FINGER || levt.code == BTN_MOUSE)
+      && levt.value == 1 && (levt.type == EV_ABS || g_advancedSettings.m_TouchMouse))
+  {
+    m_mouseX = g_graphicsContext.GetWidth() + 2;
```

Comment:
Lets wait for other reviews first, before you spend more time on changes.

The general process is, create a new commit in the same pr with any changes requested and once it's ready it can be squished before merge.